modules/services/window-managers/i3.nix

```diff
@@ -0,0 +1,532 @@
+        description = "Border color.";
+      };
+
+      childBorder = mkOption {
+        type = types.string;
+        description = "Child border color.";
+      };
+
+      background = mkOption {
+        type = types.string;
+        description = "Background color.";
+      };
+
+      text = mkOption {
+        type = types.string;
+        description = "Text color.";
+      };
+
+      indicator = mkOption {
+        type = types.string;
+        description = "Split indicator color.";
+      };
+    };
+  };
+
+  barModule = types.submodule {
+    options = {
+      mode = mkOption {
+        type = types.enum [ "dock" "hide" "invisible" ];
+        default = "dock";
+        description = "Bar visibility mode.";
+      };
+
+      hiddenState = mkOption {
+        type = types.enum [ "hide" "show" ];
+        default = "hide";
+        description = "The default bar mode when 'bar.mode' == 'hide'.";
+      };
+
+      position = mkOption {
+        type = types.enum [ "top" "bottom" ];
+        default = "bottom";
+        description = "The edge of the screen i3bar should show up.";
+      };
+
+      workspaceButtons = mkOption {
+        type = types.bool;
+        default = true;
+        description = "Whether workspace buttons should be shown or not.";
+      };
+
+      statusCommand = mkOption {
+        type = types.string;
+        default = "${pkgs.i3status}/bin/i3status";
+        description = "Command that will be used to get status lines.";
+      };
+
+    };
+  };
+
+  criteriaModule = types.attrs;
+
+  configModule = types.submodule {
+    options = {
+      fonts = mkOption {
+        type = types.listOf types.string;
+        default = ["pango:monospace 8"];
+        description = ''
+          Font list used for window titles. Only FreeType fonts are supported.
+          The order here is improtant (e.g. icons font should go before the one used for text).
+        '';
+        example = [ "FontAwesome 10" "Terminus 10" ];
+      };
+
+      window = mkOption {
+        type = types.submodule {
+          options = {
+            titlebar = mkOption {
+              type = types.bool;
+              default = cfg.package != pkgs.i3-gaps;
+              defaultText = "cfg.package != pkgs.i3-gaps (titlebar should be disabled for i3-gaps)";
+              description = "Whether to show window titlebars.";
+            };
+            border = mkOption {
+              type = types.int;
+              default = 2;
+              description = "Window border width.";
+            };
+          };
+        };
+        default = {};
+        description = "Window titlebar and border settings.";
+      };
+
+      floating = mkOption {
+        type = types.submodule {
+          options = {
+            titlebar = mkOption {
+              type = types.bool;
+              default = cfg.package != pkgs.i3-gaps;
+              defaultText = "cfg.package != pkgs.i3-gaps (titlebar should be disabled for i3-gaps)";
```

Comment:
The "cfg" here isn't very understandable for a user reading the documentation. Unfortunately will need something more extensive like "xsession.wm.i3.package != pkgs.i3-gaps" (maybe even change "pkgs" to "nixpkgs"?).

Reply:
Good point!